gateway/src/apicast/mapping_rule.lua

```diff
@@ -109,4 +114,19 @@ function _M.from_proxy_rule(proxy_rule)
   )
 end
 
+--- Checks if the mapping rule matches a given request method, path, and args
+--
+-- @tparam string method HTTP method (GET, POST, etc.).
+-- @tparam string path Path of an HTTP request.
+-- @tparam table args Table with the args and values of an HTTP request.
+-- @treturn boolean Whether the mapping rule matches the given request.
+function _M:matches(method, path, args)
```

Comment:
If we introduce SOAP rules that are full URL the `path` will no longer be just path, but a full URI. But a `pathj` is an `URI`. So maybe lets call it `uri` ?

Reply:
Good point.

Reply:
Changed :+1: